util/src/main/java/io/grpc/util/OutlierDetectionLoadBalancer.java

```diff
@@ -211,8 +211,16 @@ protected Helper delegate() {
     public Subchannel createSubchannel(CreateSubchannelArgs args) {
       // Subchannels are wrapped so that we can monitor call results and to trigger failures when
       // we decide to eject the subchannel.
-      OutlierDetectionSubchannel subchannel = new OutlierDetectionSubchannel(
-          delegate.createSubchannel(args));
+      LoadBalancer.SubchannelStateListener healthConsumerListener =
+          args.getOption(HEALTH_CONSUMER_LISTENER_ARG_KEY);
+      OutlierDetectionSubchannel subchannel =
+          new OutlierDetectionSubchannel(healthConsumerListener);
```

Comment:
Couldn't you add both the delegate and the hc listener in the constructor of `OutlierDetectionSubchannel`? That way you wouldn't have to add that extra `setDelegate()` method?

Reply:
Yea this is the workaround to break the dependency:
Original Subchannel --> `CreateSubchannelArgs`  --> `OutlierDetectionSubchannel` --> Original subchannel

Reply:
Ok, I think I see it, but it's kind of messy, is this something we can eventually clean up?

Reply:
hm, i guess i can do it another way to make it more enclosed.